### ISSUE TITLE
Add `iter_geometry` and `iter_prop` to Geojsonm

### ISSUE THREAD

**Issue by patricoferris** [geocaml/ocaml-geojson#7]
Currently Geojsonm supports mapping and folding over geometries and properties and it would be good to add equivalent `iter` functions that look something like:

```ocaml
val iter_geometry : (G.Geometry.t -> unit) -> Jsonm.src -> unit
```

**Comment by SHIVANGISINGH1** [comment]
Hello @patricoferris sir
I would like to work on this, please assign this issue to me.

Thank you

**Comment by patricoferris** [comment]
Hi @SHIVANGISINGH1 -- thanks for wanting to work on this issue, I have assigned you now :))

**Comment by IIITM-Jay** [comment]
Hi Sir, @patricoferris , while going through the OCaml libraries, I got to know that `map` and `iter` functions very similar to each other. I am not able to get the exact difference between those.

Hence, if you have some time can you guide me what's the difference between the two, why there is a need of adding this feature and a bit on how can we approach to implement this....

As Sir, now I am working on this , so looking forward to hear from you regarding this issue.

**Comment by IIITM-Jay** [comment]
As far as I am guessing, we have to change these lines from `map` function:
 
https://github.com/geocaml/ocaml-geojson/blob/27ffcf9f7de40cd73295289fb5d3451ad628b75f/src/geojsonm/geojsonm.ml#L175

https://github.com/geocaml/ocaml-geojson/blob/27ffcf9f7de40cd73295289fb5d3451ad628b75f/src/geojsonm/geojsonm.ml#L180

Hi @patricoferris, may I request you to kindly throw some light on above..

**Comment by SHIVANGISINGH1** [comment]
> 

Hey @patricoferris so sorry to reply late 🙏 for the past few days I was busy with my exams and I'll be busy for a couple more till they are over 😓.  I have worked on this issue before but I'll have to give it some more time so for the time being, I think you can assign it to @IIITM-Jay or in the upcoming week as I'll be free we both can work on the issue together. 

Thank you 🙇‍♀️

**Comment by patricoferris** [comment]
Hi @SHIVANGISINGH1 :))

Good luck with your exams! I appreciate your enthusiasm, but I did have to let the issue be solved by someone else given it was a week that you had it. I know how exams can be a little all-consuming, so don't worry about it. Let me know if you get more free time soon and I can open an issue for you so that you have a chance to contribute.

**Comment by SHIVANGISINGH1** [comment]
> Hi @SHIVANGISINGH1 :))
> 
> Good luck with your exams! I appreciate your enthusiasm, but I did have to let the issue be solved by someone else given it was a week that you had it. I know how exams can be a little all-consuming, so don't worry about it. Let me know if you get more free time soon and I can open an issue for you so that you have a chance to contribute.

Thank you so much for your support @patricoferris , as I am free with my exams now I will take some time to understand OCaml and geojson and get familiar with the codebase. After that, I would love to contribute to an issue. ☺✌

**Comment by patricoferris** [comment]
Fixed in #25
